src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -619,7 +626,14 @@ private async Task ProcessSearchResultsAsync(IEnumerable<SearchResultEntry> sear
 
                 if (anonymizer != null)
                 {
-                    element = anonymizer.Anonymize(element);
+                    try
+                    {
+                        element = anonymizer.Anonymize(element);
+                    }
+                    catch (Exception ex)
+                    {
+                        throw new FailedToAnonymizeResourceException(ex.Message, ex);
```

Comment:
Should we log the error and continue anonymize other resource?

Reply:
Currently we would like to fail fast if any unexpected exception happen.